mempool.go

```diff
@@ -1132,6 +1132,25 @@ func (mp *txMemPool) maybeAcceptTransaction(tx *btcutil.Tx, isNew, rateLimit boo
 		return nil, txRuleError(wire.RejectInsufficientFee, str)
 	}
 
+	// Require that free transactions have sufficient priority to be mined
+	// in the next block.
+	if !cfg.NoRelayPriority {
+		txD := &TxDesc{
+			Tx:     tx,
+			Added:  time.Now(),
+			Height: curHeight,
+			Fee:    txFee,
+		}
+		if txFee < minFee &&
```

Comment:
I'd move this first part of the check into the if statement above, to avoid creating the TxDesc when this is false.